fleetshard/pkg/runtime/runtime.go

```diff
@@ -107,6 +107,13 @@ func (r *Runtime) Start() error {
 		Telemetry:         r.config.Telemetry,
 	}
 
+	glog.Infof("Loading cluster configuration")
+	_, _, err := r.client.PrivateAPI().GetDataPlaneClusterAgentConfig(context.Background(), r.clusterID)
+	// Sanity check if the cluster configuration is correct.
+	if err != nil {
```

Comment:
I don't feel strongly about this, but how about:
```suggestion
	// Sanity check if the cluster configuration is correct.
	if _, _, err := r.client.PrivateAPI().GetDataPlaneClusterAgentConfig(context.Background(), r.clusterID); err != nil {
```

Reply:
Also, shouldn't the context have a deadline? :thinking:

Reply:
Done.